### DISCUSSION TITLE
Best DNS config for me?

### DISCUSSION THREAD

**Discussion by Fr3d3rick83** [AdguardTeam/AdGuardHome#4536] [Q&A]
Hi, I´m using AdGuard Home on my Synology NAS via Docker. How can I figure out, what is the best and fastest configuration for the DNS settings?

Thanks
Frederick

**Comment by Fr3d3rick83** [comment]
Isnt there any suggsestion, since everything loads a bit slower and in the game Destiny 2, I often get connection losts?

**Reply by joshqou** [reply]
If you're having connection issues in an online game, that's more likely to be the fault of your internet connection than AdGuard Home. The most DNS would be used for would be when initially connecting to a gameserver.

You should check the query log to see the processing time for cached and uncached requests, that should help narrow down if the slowness is the cause of your ISP or your NAS being overworked.

Enabling caching in *DNS Settings* can help if your processing time for uncached responses is particularly high. If that's the case you might also want to [change your upstream dns servers](https://kb.adguard.com/general/dns-providers) to ones geographically closer to you

**Reply by Fr3d3rick83** [reply]
Thanks for you anwser.
What are good cache values and how can I see, what DNS server is close to me, I thought that Cloudflare as a wide spread. I also read, that its good to have several IPs listed, is that true? I added a screenshot of my settings to get a bether view.
![Screenshot 2022-05-10 at 09-01-56 AdGuard Home](https://user-images.githubusercontent.com/97901424/167568270-8d3a2fe3-faab-4706-9762-a307c240ffc6.png)

**Reply by joshqou** [reply]
While I'm sure there are sites that can provide you a list of regional DNS providers, you can also plug some of the providers that interest you into `ping` to see how fast your connection to them is. I'd probably try messing with the upstream servers the last though

You should look at the query log first before messing with any of your dns settings

> I also read, that its good to have several IPs listed, is that true?

For parallel mode you probably shouldn't have too many as you'll be causing AGH to send out 8 or so DNS queries for every one DNS query one of your devices tries to make, which could slow your network a bit if you have a lot of devices and low bandwidth. 3-4 upstream servers should be more than enough

> What are good cache values and how can I see

You can see your cache values under the "DNS-Cache" section. Your existing cache settings should be fine provided your NAS isn't ram-limited. Turning on optimistic caching might help with stale cache results

**Reply by Fr3d3rick83** [reply]
Ok, the processing time is between 22 and 47ms, the average processing time is 30ms.
![Bildschirmfoto 2022-05-10 um 09 53 19](https://user-images.githubusercontent.com/97901424/167577825-c5e77b26-6fa5-4c1c-bd61-462825e03060.png)

So should I switch to "Load balacing" and not to "Parallel"?

![Bildschirmfoto 2022-05-10 um 10 01 55](https://user-images.githubusercontent.com/97901424/167579321-1cd548ba-1ac5-4424-a1d1-484bec11e695.png)
![Bildschirmfoto 2022-05-10 um 10 01 27](https://user-images.githubusercontent.com/97901424/167579327-5700b8cb-93be-4597-9678-3ecc77cf722e.png)
![Bildschirmfoto 2022-05-10 um 10 01 35](https://user-images.githubusercontent.com/97901424/167579332-b18918d1-09bd-4850-b6f7-6225c141c7ec.png)
![Bildschirmfoto 2022-05-10 um 10 01 42](https://user-images.githubusercontent.com/97901424/167579334-d64c6db7-be60-47a5-8f7f-68bb7a6291eb.png)

For what I saw the 1.0.0.1 DNS is the fastest, all the other in my list, take longer.

**Reply by joshqou** [reply]
Did none of those queries hit your cache? If you've got an iPhone or Mac on your network it should be chatty enough to keep any Apple-related domains cached

> So should I switch to "Load balacing" and not to "Parallel"?

Either mode should work fine granted you're using fast upstream servers

**Comment by Fr3d3rick83** [comment]
I changed the upstream server to: 
1.0.0.1
1.1.1.1

How do I see, if a page is cached or not?

I changed the DNS Rate limit to "0", since all request come from my router.
And I deselected all checkboxes.

**Reply by joshqou** [reply]
![167592938-8f3cd87d-61dd-4eb7-8682-5fa4c3cabdcf.png](https://user-images.githubusercontent.com/97894002/167592938-8f3cd87d-61dd-4eb7-8682-5fa4c3cabdcf.png)

Cached responses will have a *(served from cache)* note next to the upstream DNS server that served the response.

**Reply by Fr3d3rick83** [reply]
Yes, since I checked the "optimise caching", some get cached. But how do you get 1ms?...WOW!

**Reply by joshqou** [reply]
I'm using a Raspberry Pi 3 that only runs AGH. Raspbian likely runs lighter than the firmware used on your nas, meaning AGH isn't fighting for processing time.

Your cached responses should be very quick compared to queries that miss the cache. For me it's <5ms hit <30ms miss

**Comment by Fr3d3rick83** [comment]
To bad, so are there some other optimisations I can do?

**Reply by joshqou** [reply]
You could try running AGH on a dedicated low power machine like a RPi, or set up AGH to run outside of docker to get rid of the small overhead docker has. Anything beyond tweaking settings in the AGH UI will be a lot more involved however.

If you have ssh access to your nas, you should run `top` to see what else your nas has running in the background. There are likely some services or features it has enabled in the background you could turn off if you don't use them
